tools/bazel.rc

```diff
@@ -1,7 +1,3 @@
-build --copt -W
-build --copt -Wall
```

Comment:
@thibault-martinez I'm okay with it, do you agree to remove these warning options?

Reply:
@tsvisabo Why are we removing them ?

Reply:
I don't have any third party related warnings, I don't think we should remove them.

Reply:
Because boringssl though external fails with these flags due to unused params
@thibault-martinez has found that boringssl define -Werror and it collides with our -Wextra
so one solution is of course to remove those flags, another one is to try and suppress flags from dependency though i don't see how

Reply:
I did some test on this PR. it's not related to `boringssl`'s warning options, these compile options only apply to itself, it won't affect the whole project. 
we can keep `-Wall -Wextra`, remove `-W` which is redundant and `-Wno-error=unused-parameter -Wno-error=cast-function-type` they are cause compiling failed on `googletest`, `curl-p`, `trinary` and so on in my env. 
adding these options causes incompatible issues between compilers. we should support different compilers, IMO.

Reply:
> it's not related to boringssl's warning options, these compile options only apply to itself, it won't affect the whole project.

Yes but it's the other way around. We are providing `-Wextra` that is propagated to BoringSSL which has `-Werror` hence the failure. Doing what you say still ends up in compilation error.

Reply:
And how can `googletest`, `curl-p` and `trinary` fail if we don't have a `-Werror` on our end ? Can you send the actual error ? Or do you mean that `-Wno-error=unused-parameter` and  `-Wno-error=cast-function-type` simply doesn't exist ?

Reply:
> And how can `googletest`, `curl-p` and `trinary` fail if we don't have a `-Werror` on our end ? Can you send the actual error ? Or do you mean that `-Wno-error=unused-parameter` and `-Wno-error=cast-function-type` simply doesn't exist on another compiler ?

gcc 7.4.0 has no `cast-function-type` option.

```
ERROR: /home/sam/.cache/bazel/_bazel_sam/6033e706b65718d7c9e2770fdaa44559/external/com_google_googletest/BUILD.bazel:55:1: C++ compilation of rule '@com_google_googletest//$
gtest' failed (Exit 1) gcc failed: error executing command /usr/bin/gcc -U_FORTIFY_SOURCE -fstack-protector -Wall -Wunused-but-set-parameter -Wno-free-nonheap-object -fno-o$
it-frame-pointer '-std=c++0x' -MD -MF ... (remaining 38 argument(s) skipped)

cc1plus: error: -Werror=cast-function-type: no option -Wcast-function-type
```

Reply:
> > it's not related to boringssl's warning options, these compile options only apply to itself, it won't affect the whole project.
> 
> Yes but it's the other way around. We are providing `-Wextra` that is propagated to BoringSSL which has `-Werror` hence the failure. Doing what you say still ends up in compilation error.

I wasn't thinking clearly in the middle of the night. :P
alright, `-Wextra` is apply to boringssl, so we just remove the `-Werror` in boringssl.
https://github.com/iotaledger/rules_iota/pull/46